### PR TITLE
extended standard configuration

### PR DESCRIPTION
**Pull request by pseyfert**
- load compinit for tab completion
- menu completion for long lists of
- disable beeping for tab completion

### CODE REVIEW

File: conf/20-look_and_feel [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# http://stackoverflow.com/questions/171563/whats-in-your-zshrc
```
Comment (Merovius) [inline conf/20-look_and_feel]: I will burn in FOSS-hell for this, but I don't think such a trivial line of configuration deserves attribution. If it makes you feel any better, it's in my zsh too.

Add a vim-modeline, © line and a short explanation of what that file does instead (see other files in conf/)

Comment (Merovius) [inline conf/20-look_and_feel]: Also, look_and_feel is a wrong name. Call the file "nobeep" or something, as stupid as it seems to have a single line as a separate file, but the config-style of shellex is to have isolated units that can be ln'ed.

File: conf/20-completion [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+autoload -U compinit
+compinit
+
+setopt +o bashautolist
+setopt +o listambiguous
+setopt +o autolist
+
+# these two don't seem helpful
+#setopt +o menucomplete
+#setopt +o automenu
```
Comment (Merovius) [inline conf/20-completion]: If they are not helpful, they should be deleted.

File: conf/20-completion [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# load tab completion stuff http://www.refining-linux.org/archives/40/ZSH-Gem-5-Menu-selection/
+autoload -U compinit
+compinit
+
+setopt +o bashautolist
+setopt +o listambiguous
+setopt +o autolist
```
Comment (Merovius) [inline conf/20-completion]: IIUC, autolist is redundant, if bashautolist is set.

File: conf.mk [hunk 4]
```diff
@@ -1,6 +1,6 @@
 INSTALL_TARGETS += install-conf
 
-default_confs := 10-autoexec 40-escape 40-home_end 40-setprompt 40-sigint 90-hist 99-clear
+default_confs := 10-autoexec 20-completion 20-look_and_feel 40-escape 40-home_end 40-setprompt 40-sigint 90-hist 99-clear
```
Comment (Merovius) [inline conf.mk]: I don't think the completion-thing should be installed by default. It's highly dependent on personal taste and I can't even seem to find a setting I personally like…

File: conf/20-completion [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+# In the form `yes=num', where `yes' may be any of the true values (`yes',
+# `true', `on' and `1'), menu completion will be turned on if there are at least
+# num matches.
+#
+# In the form `select=num', menu selection will only be started if there are at
+# least num matches.
+
+#neither of the three doesn't go to top. and doesn't do highlighting
+#zstyle ':completion:*:default' menu yes=0          # BAD: doesn't show ambiguity before picking the first, and no highlighting
+#zstyle ':completion:*:default' menu select=0 yes=0 # BAD: doesn't show ambiguity before picking the first
```
Comment (Merovius) [inline conf/20-completion]: If they are BAD, delete them.

**Comment by Merovius** [comment]
I have to say, I don't like the particular set of options you chose, personally and I don't feel comfortable shipping something as a default (or even the "one recommendation") that I don't consider good behavior. In around 5 minutes of tinkering I didn't end up with a behavior I like. So I don't think I want to merge this just yet.

How autocompletion should behave seems to be a highly subjective thing, which is why zsh built it's whole completion system around it and made it configurable with a gazillion config variables. I'm not sure we're able to come up with a reasonable default at all.

**Comment by Merovius** [comment]
I thought about it some more and came to the conclusion that this is not something I want in the main repository. The nobeep-file is fine and a good default, but I consider the exact configuration of zshcomp too subjective to fit well.

It was always the plan to have a community-maintained repository of snippets, so I'd invite you to start a Wiki-page (github has these wikis… never used them…) for those and put examples for zshcomp configuration there.

Feel free to either modify this pull-request to only include the nobeep-change, or close and abandon it.

**Comment by pseyfert** [comment]
Thinking about your comments, I agree. will remove the tab-completion stuff from the PR. shipping the zshcomp as community snippet is fine with me (though haven't used github wiki either).

**Comment by pseyfert** [comment]
just tested the behaviour of removing 20-completion I see what's described in #30 (flickering when tab completing in a directory with more files than fit on the screen).

But, when merging #41, I can safely remove 20-completion.

I.e. either #39 (with 20-completion) or #41 fix #30.

**Comment by pseyfert** [comment]
decided to merge no-beep and fix #30 with #41 and move my tab preferences to pseyfert/shellex-customizations.

**Pull request closed without merging**